lib/core/src/sync/mod.rs

```diff
@@ -39,6 +43,47 @@ impl SyncService {
         }
     }
 
+    fn check_remote_change(&self) -> Result<()> {
+        match self
+            .persister
+            .get_sync_settings()?
+            .remote_url
+            .is_some_and(|url| url == self.remote_url)
+        {
+            true => Ok(()),
+            false => self.persister.set_new_remote(self.remote_url.clone()),
+        }
+    }
+
+    pub(crate) async fn run(self: Arc<Self>, mut shutdown: watch::Receiver<()>) -> Result<()> {
```

Comment:
Shall we call it `start()` for consistency?

Reply:
Handled in next PR (integrate)